remotes/docker/config/hosts.go

```diff
@@ -0,0 +1,481 @@
+					if options.DefaultScheme != "" {
+						hosts[len(hosts)-1].scheme = options.DefaultScheme
+					} else {
+						hosts[len(hosts)-1].scheme = "https"
+					}
+				}
+				hosts[len(hosts)-1].path = "/v2"
+			}
+		}
+
+		defaultTransport := &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			MaxIdleConns:          10,
+			IdleConnTimeout:       30 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			TLSClientConfig:       options.DefaultTLS,
+			ExpectContinueTimeout: 5 * time.Second,
+		}
+
+		client := &http.Client{
+			Transport: defaultTransport,
+		}
+
+		authOpts := []docker.AuthorizerOpt{docker.WithAuthClient(client)}
+		if options.Credentials != nil {
+			authOpts = append(authOpts, docker.WithAuthCreds(options.Credentials))
+		}
+		authorizer := docker.NewDockerAuthorizer(authOpts...)
+
+		rhosts := make([]docker.RegistryHost, len(hosts))
+		for i, host := range hosts {
+
+			rhosts[i].Scheme = host.scheme
+			rhosts[i].Host = host.host
+			rhosts[i].Path = host.path
+			rhosts[i].Capabilities = host.capabilities
+
+			if host.caCerts != nil || host.clientPairs != nil || host.skipVerify != nil {
+				var tlsConfig *tls.Config
+				if options.DefaultTLS != nil {
+					tlsConfig = options.DefaultTLS.Clone()
+				} else {
+					tlsConfig = &tls.Config{}
+				}
+				if host.skipVerify != nil {
+					tlsConfig.InsecureSkipVerify = *host.skipVerify
+				}
+				if host.caCerts != nil {
+					if tlsConfig.RootCAs == nil {
+						rootPool, err := rootSystemPool()
+						if err != nil {
+							return nil, errors.Wrap(err, "unable to initialize cert pool")
+						}
+						tlsConfig.RootCAs = rootPool
+					}
+					for _, f := range host.caCerts {
+						data, err := ioutil.ReadFile(f)
+						if err != nil {
+							return nil, errors.Wrapf(err, "unable to read CA cert %q", f)
+						}
+						if !tlsConfig.RootCAs.AppendCertsFromPEM(data) {
+							return nil, errors.Errorf("unable to load CA cert %q", f)
+						}
+					}
+				}
+
+				if host.clientPairs != nil {
+					for _, pair := range host.clientPairs {
+						certPEMBlock, err := ioutil.ReadFile(pair[0])
+						if err != nil {
+							return nil, errors.Wrapf(err, "unable to read CERT file %q", pair[0])
+						}
+						var keyPEMBlock []byte
+						if pair[1] != "" {
+							keyPEMBlock, err = ioutil.ReadFile(pair[1])
+							if err != nil {
+								return nil, errors.Wrapf(err, "unable to read CERT file %q", pair[1])
+							}
+						} else {
+							// Load key block from same PEM file
+							keyPEMBlock = certPEMBlock
+						}
+						cert, err := tls.X509KeyPair(certPEMBlock, keyPEMBlock)
+						if err != nil {
+							return nil, errors.Wrap(err, "failed to load X509 key pair")
+						}
+
+						tlsConfig.Certificates = append(tlsConfig.Certificates, cert)
+					}
+				}
+				tr := defaultTransport.Clone()
```

Comment:
Looks like this makes it break on Go < 1.13;

```bash
go get golang.org/dl/go1.12.17 && go1.12.17 download

go1.12.17 run cmd/gen-manpages/main.go ctr man/
# github.com/containerd/containerd/remotes/docker/config
../../go/src/github.com/containerd/containerd/remotes/docker/config/hosts.go:193:27: defaultTransport.Clone undefined (type *http.Transport has no field or method Clone)
```

Reply:
1.12 reached EOL